common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -802,3 +785,44 @@ describe 'Problem', ->
 
       # verify that codemirror textarea has correct `aria-describedby` attribute value
       expect($(CodeMirrorTextArea).attr('aria-describedby')).toEqual('cm-editor-exit-message-101 status_101')
+
+
+  describe 'show answer button', ->
+
+    radioButtonProblemHtml = readFixtures('radiobutton_problem.html')
+    checkboxProblemHtml = readFixtures('checkbox_problem.html')
+
+    beforeEach ->
+      @problem = new Problem($('.xblock-student_view'))
+
+    it 'should become enabled after a radiobutton is selected', ->
+      $('#input_example_1').replaceWith(radioButtonProblemHtml)
+
+      $('#input_1_1_1').attr('checked', true)
+      @problem.el.find('.show').trigger('click')
+      @problem.el.find('div.choicegroup label:nth-child(2)').addClass('choicegroup_correct')
+      @problem.el.find('div.choicegroup .choicegroup_correct').append('<span class="status correct"></span>')
+      @problem.el.find('.show').attr('disabled', 'disabled')
+
+      @problem.submitAnswersAndSubmitButton true
+
+      $('#input_1_1_2').attr('checked', true).trigger('click')
+      expect(@problem.el.find('.show').attr('disabled')).not.toEqual('disabled')
+      expect(@problem.el.find('div.choicegroup')).not.toHaveClass('choicegroup_correct')
+      expect(@problem.el.find('div.choicegroup')).not.toHaveAttr('span.status.correct')
+
+    it 'should become enabled after a checkbox is selected', ->
+      $('#input_example_1').replaceWith(checkboxProblemHtml)
+
+      $('#input_1_1_1').attr('checked', true)
```

Comment:
Starting from here to line # 821 is same as from line # 801-805. What can we do to remove this repetition?